// docx/build/docx/document/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const paragraph_1 = require("../paragraph");
const table_1 = require("../table");
const xml_components_1 = require("../xml-components");
const body_1 = require("./body");
const document_attributes_1 = require("./document-attributes");
class Document extends xml_components_1.XmlComponent {
    constructor() {
        super("w:document");
        this.root.push(new document_attributes_1.DocumentAttributes({
            wpc: "http://schemas.microsoft.com/office/word/2010/wordprocessingCanvas",
            mc: "http://schemas.openxmlformats.org/markup-compatibility/2006",
            o: "urn:schemas-microsoft-com:office:office",
            r: "http://schemas.openxmlformats.org/officeDocument/2006/relationships",
            m: "http://schemas.openxmlformats.org/officeDocument/2006/math",
            v: "urn:schemas-microsoft-com:vml",
            wp14: "http://schemas.microsoft.com/office/word/2010/wordprocessingDrawing",
            wp: "http://schemas.openxmlformats.org/drawingml/2006/wordprocessingDrawing",
            w10: "urn:schemas-microsoft-com:office:word",
            w: "http://schemas.openxmlformats.org/wordprocessingml/2006/main",
            w14: "http://schemas.microsoft.com/office/word/2010/wordml",
            w15: "http://schemas.microsoft.com/office/word/2012/wordml",
            wpg: "http://schemas.microsoft.com/office/word/2010/wordprocessingGroup",
            wpi: "http://schemas.microsoft.com/office/word/2010/wordprocessingInk",
            wne: "http://schemas.microsoft.com/office/word/2006/wordml",
            wps: "http://schemas.microsoft.com/office/word/2010/wordprocessingShape",
            Ignorable: "w14 w15 wp14",
        }));
        this.body = new body_1.Body();
        this.root.push(this.body);
    }
    addParagraph(paragraph) {
        this.body.push(paragraph);
    }
    createParagraph(text) {
        const para = new paragraph_1.Paragraph(text);
        this.addParagraph(para);
        return para;
    }
    addTable(table) {
        this.body.push(table);
    }
    createTable(rows, cols) {
        const table = new table_1.Table(rows, cols);
        this.addTable(table);
        return table;
    }
}
exports.Document = Document;
