// docx/build/styles/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const document_attributes_1 = require("../docx/document/document-attributes");
const xml_components_1 = require("../docx/xml-components");
const style_1 = require("./style");
class Styles extends xml_components_1.XmlComponent {
    constructor() {
        super("w:styles");
        this.root.push(new document_attributes_1.DocumentAttributes({
            mc: "http://schemas.openxmlformats.org/markup-compatibility/2006",
            r: "http://schemas.openxmlformats.org/officeDocument/2006/relationships",
            w: "http://schemas.openxmlformats.org/wordprocessingml/2006/main",
            w14: "http://schemas.microsoft.com/office/word/2010/wordml",
            w15: "http://schemas.microsoft.com/office/word/2012/wordml",
            Ignorable: "w14 w15",
        }));
    }
    push(style) {
        this.root.push(style);
        return this;
    }
    createParagraphStyle(styleId, name) {
        const para = new style_1.ParagraphStyle(styleId, name);
        this.push(para);
        return para;
    }
}
exports.Styles = Styles;
