// docx/build/docx/run/formatting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const xml_components_1 = require("../xml-components");
var underline_1 = require("./underline");
exports.Underline = underline_1.Underline;
var script_1 = require("./script");
exports.SubScript = script_1.SubScript;
exports.SuperScript = script_1.SuperScript;
var run_fonts_1 = require("./run-fonts");
exports.RunFonts = run_fonts_1.RunFonts;
class Bold extends xml_components_1.XmlComponent {
    constructor() {
        super("w:b");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Bold = Bold;
class Italics extends xml_components_1.XmlComponent {
    constructor() {
        super("w:i");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Italics = Italics;
class Caps extends xml_components_1.XmlComponent {
    constructor() {
        super("w:caps");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Caps = Caps;
class Color extends xml_components_1.XmlComponent {
    constructor(color) {
        super("w:color");
        this.root.push(new xml_components_1.Attributes({
            val: color,
        }));
    }
}
exports.Color = Color;
class DoubleStrike extends xml_components_1.XmlComponent {
    constructor() {
        super("w:dstrike");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.DoubleStrike = DoubleStrike;
class Emboss extends xml_components_1.XmlComponent {
    constructor() {
        super("w:emboss");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Emboss = Emboss;
class Imprint extends xml_components_1.XmlComponent {
    constructor() {
        super("w:imprint");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Imprint = Imprint;
class Shadow extends xml_components_1.XmlComponent {
    constructor() {
        super("w:shadow");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Shadow = Shadow;
class SmallCaps extends xml_components_1.XmlComponent {
    constructor() {
        super("w:smallCaps");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.SmallCaps = SmallCaps;
class Strike extends xml_components_1.XmlComponent {
    constructor() {
        super("w:strike");
        this.root.push(new xml_components_1.Attributes({
            val: true,
        }));
    }
}
exports.Strike = Strike;
class Size extends xml_components_1.XmlComponent {
    constructor(size) {
        super("w:sz");
        this.root.push(new xml_components_1.Attributes({
            val: size,
        }));
    }
}
exports.Size = Size;
