// docx/build/docx/paragraph/formatting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var alignment_1 = require("./alignment");
exports.Alignment = alignment_1.Alignment;
var border_1 = require("./border");
exports.ThematicBreak = border_1.ThematicBreak;
var indent_1 = require("./indent");
exports.Indent = indent_1.Indent;
var keep_1 = require("./keep");
exports.KeepLines = keep_1.KeepLines;
exports.KeepNext = keep_1.KeepNext;
var page_break_1 = require("./page-break");
exports.PageBreak = page_break_1.PageBreak;
var properties_1 = require("./properties");
exports.ParagraphProperties = properties_1.ParagraphProperties;
var spacing_1 = require("./spacing");
exports.Spacing = spacing_1.Spacing;
var style_1 = require("./style");
exports.Style = style_1.Style;
var tab_stop_1 = require("./tab-stop");
exports.LeftTabStop = tab_stop_1.LeftTabStop;
exports.MaxRightTabStop = tab_stop_1.MaxRightTabStop;
var unordered_list_1 = require("./unordered-list");
exports.NumberProperties = unordered_list_1.NumberProperties;
