// docx/build/docx/paragraph/tab-stop.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const xml_components_1 = require("../xml-components");
class TabStop extends xml_components_1.XmlComponent {
    constructor(tab) {
        super("w:tabs");
        this.root.push(tab);
    }
}
exports.TabStop = TabStop;
class TabAttributes extends xml_components_1.XmlAttributeComponent {
    constructor() {
        super(...arguments);
        this.xmlKeys = { val: "w:val", pos: "w:pos" };
    }
}
exports.TabAttributes = TabAttributes;
class Tab extends xml_components_1.XmlComponent {
    constructor(value, position) {
        super("w:tab");
        this.root.push(new TabAttributes({
            val: value,
            pos: position,
        }));
    }
}
exports.Tab = Tab;
class MaxRightTabStop extends TabStop {
    constructor() {
        super(new Tab("right", 9026));
    }
}
exports.MaxRightTabStop = MaxRightTabStop;
class LeftTabStop extends TabStop {
    constructor(position) {
        super(new Tab("left", position));
    }
}
exports.LeftTabStop = LeftTabStop;
